### PR TITLE
Enabling the button after create scan failure (AST-77386)

### PR DESCRIPTION
**Pull request by sarahCx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

Catching the error returned from CX and clearing scan-related states.

### References

https://checkmarx.atlassian.net/browse/AST-77386

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: src/views/resultsView/createScanProvider.ts [hunk 1]
```diff
@@ -153,8 +154,12 @@ export async function createScan(
   }
 
   updateStatusBarItem(constants.scanCreatePreparing, true, statusBarItem);
-  await createScanForProject(context, logs);
-
+  try {
+    await createScanForProject(context, logs);
+  } catch (error) {
+      setScanButtonDefaultIfScanIsNotRunning(context);
```
Comment (miryamfoiferCX) [inline src/views/resultsView/createScanProvider.ts:160]: what happen if scan is running?
maybe we want to add a log with this information to notify the user abut stack scan?

Comment (miryamfoiferCX) [inline src/views/resultsView/createScanProvider.ts:160]: can we add a test to this flow? (UT is enough)

Comment (sarahCx) [reply src/views/resultsView/createScanProvider.ts:160]: The run scan button will be a cancel button.
the user get a notification with the error (I keep throwing the error)

Comment (sarahCx) [reply src/views/resultsView/createScanProvider.ts:160]: We didn't have a UT.

Comment (miryamfoiferCX) [reply src/views/resultsView/createScanProvider.ts:160]: will method setScanButtonDefaultIfScanIsNotRunning work properly even if the button is cancel and not run?

**Review by miryamfoiferCX — APPROVED** [review]

**Pull request merged**